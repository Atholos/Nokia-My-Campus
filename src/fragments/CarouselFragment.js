


import {makeStyles} from "@material-ui/core";
import TabFragments from "./TabFragments";
import React, {Fragment, useState} from "react";
import Carousel from "react-material-ui-carousel";
import RestaurantHeatMapView from "../views/restaurantHeatMap";


const useStyle = makeStyles(() => ({
    Carousel: {
        height: "100%",
    },
    Tab: {
       height: "100%",
    },
}));

const CarouselFragment = (props) => {
    const classes = useStyle();
    const {TabRestaurantLines, TabRestaurantChart, TabRestaurantDonut, TabRestaurantMenu} = TabFragments();

    const Restaurant = () => {
        const [valueRestaurant] = useState(0);
        const [date, setDate] = useState(new Date());

        const handleDateChange = (data) => {
            setDate(data);
        };
//
        return (
                <Fragment>
                        <Carousel autoPlay={false} navButtonsAlwaysVisible={true}
                                  animation="slide" indicators={true} className={classes.Carousel}>
                            <TabRestaurantMenu value={valueRestaurant} index={0}/>
                            <TabRestaurantLines value={valueRestaurant} index={0}/>
                            <TabRestaurantChart value={valueRestaurant} index={0} onDateChange={handleDateChange} date={date}/>
                            <RestaurantHeatMapView value={valueRestaurant} index={0}/>
                            <TabRestaurantDonut value={valueRestaurant} index={0} onDateChange={handleDateChange} date={date}/>
                        </Carousel>
                </Fragment>
        );
    };

    return {
        Restaurant: Restaurant,
    };

};
export default CarouselFragment;