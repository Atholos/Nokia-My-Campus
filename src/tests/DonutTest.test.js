import React from 'react';
import {createMount, createShallow} from '@material-ui/core/test-utils'
import '@testing-library/jest-dom/extend-expect';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from "enzyme-to-json";
import Restaurant from "../views/restaurant";
import CarouselFragment from "../fragments/CarouselFragment";

Enzyme.configure({ adapter: new Adapter() });

describe('Restaurant', () =>{
    let mount;
    let shallow;

    beforeAll( () =>{
        mount = createMount();
        shallow = createShallow();
    });

    afterAll( () =>{
       mount.cleanUp();
       shallow.cleanUp();
    });

    it('renders', () =>{
        const wrapper = mount(<Restaurant/>);
        expect(wrapper).toMatchSnapshot();
    });
/*
    it('Finds a specific div', () =>{
        const wrapper = shallow(<Restaurant><CarouselFragment/></Restaurant>);
        const fragment = wrapper.instance().render();
        console.log(wrapper.debug());
        expect(shallowToJson(fragment).getElement()).toMatchSnapshot();
    });
    */
});