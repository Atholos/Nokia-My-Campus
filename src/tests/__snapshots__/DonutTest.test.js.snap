// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurant renders 1`] = `
<Restaurant>
  <AuthRestaurant>
    <AuthLoading>
      <div
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "display": "flex",
            "height": "100%",
            "justifyContent": "center",
            "left": 0,
            "margin": "auto 0",
            "position": "absolute",
            "right": 0,
            "textAlign": "center",
            "top": 0,
          }
        }
      >
        <WithStyles(ForwardRef(CircularProgress))
          style={Object {}}
        >
          <ForwardRef(CircularProgress)
            classes={
              Object {
                "circle": "MuiCircularProgress-circle",
                "circleDeterminate": "MuiCircularProgress-circleDeterminate",
                "circleDisableShrink": "MuiCircularProgress-circleDisableShrink",
                "circleIndeterminate": "MuiCircularProgress-circleIndeterminate",
                "circleStatic": "MuiCircularProgress-circleStatic",
                "colorPrimary": "MuiCircularProgress-colorPrimary",
                "colorSecondary": "MuiCircularProgress-colorSecondary",
                "determinate": "MuiCircularProgress-determinate",
                "indeterminate": "MuiCircularProgress-indeterminate",
                "root": "MuiCircularProgress-root",
                "static": "MuiCircularProgress-static",
                "svg": "MuiCircularProgress-svg",
              }
            }
            style={Object {}}
          >
            <div
              className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
              role="progressbar"
              style={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
            >
              <svg
                className="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx={44}
                  cy={44}
                  fill="none"
                  r={20.2}
                  strokeWidth={3.6}
                  style={Object {}}
                />
              </svg>
            </div>
          </ForwardRef(CircularProgress)>
        </WithStyles(ForwardRef(CircularProgress))>
      </div>
    </AuthLoading>
  </AuthRestaurant>
</Restaurant>
`;
